src/account-info/ProfileCard.js

```diff
@@ -0,0 +1,66 @@
+/* @flow strict-local */
+import React, { PureComponent } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { connect } from 'react-redux';
+
+import type { Dispatch, GlobalState, User, PresenceState } from '../types';
+import { ZulipButton } from '../common';
+import { getOwnUser, getPresence } from '../selectors';
+import { navigateToUserStatus } from '../actions';
+import AccountDetails from './AccountDetails';
+import SwitchAccountButton from './SwitchAccountButton';
+import LogoutButton from './LogoutButton';
+import AwayStatusSwitch from './AwayStatusSwitch';
+
+const componentStyles = StyleSheet.create({
+  accountButtons: {
+    alignItems: 'flex-end',
+    flexDirection: 'row',
+    marginHorizontal: 8,
+  },
+});
+
+type PropsFromState = {|
+  ownUser: User,
+  presence: PresenceState,
+|};
+
+type Props = {|
+  ...PropsFromState,
+  dispatch: Dispatch,
+|};
+
+class ProfileCard extends PureComponent<Props> {
+  handleSetUserStatus = () => {
+    const { dispatch } = this.props;
+    dispatch(navigateToUserStatus());
+  };
+
+  handleUpdateAwayStatus = () => {};
+
+  render() {
+    const { ownUser, presence } = this.props;
+
+    return (
+      <View>
+        <AccountDetails user={ownUser} presence={presence[ownUser.email]} />
```

Comment:
You'll want to simplify this when rebasing atop your more recent refactor that nicely did that for existing components.